### PR TITLE
RDM-1662: Create Dropdown and other UI fixes

### PR DESCRIPTION
**Pull request by MKaramath**
https://tools.hmcts.net/jira/browse/RDM-1662

**_Note_**: This PR should be merge along with the below PR https://github.com/hmcts/ccd-case-management-web/pull/117

### Change description ###
The dropdown fields to conform with UX guidelines and behaviour.   This change is applied on all pages where dropdown field exists. 

Other UI changes:
- Responsive is applied on Case list, Create case pages
- Dropdown styling is corrected to guidelines on all browsers
- On case list, the spaces surrounding the case list filters is fixed with responsiveness; the left and right alignments of filter results are corrected.
- On create case, headings (H1, H2, H3), field labels, label hints, and all others fields styling are corrected to conform the UX guidelines
- On search filter, the dropdowns are updated as per guidelines
- **£** symbol of money input field is now placed fixed inside the text as per guidelines
- Read only email field on the wizards pages is enabled to a link
- The changes related to the PR https://github.com/hmcts/ccd-case-management-web/pull/117 is now merged into this PR due to the UI layout dependency.
- The read only email field is enabled as link to open a blank email addressing to the email (refer https://tools.hmcts.net/jira/browse/RDM-2309)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/app/shared/case-editor/case-edit-submit.component.ts [hunk 1]
```diff
@@ -223,4 +223,20 @@ export class CaseEditSubmitComponent implements OnInit {
       .filter(cf => cf.show_summary_content_option);
   }
 
+  getCaseId(): String {
+    return (this.caseEdit.caseDetails ? this.caseEdit.caseDetails.case_id : '');
+  }
+
+  isEmail(field: CaseField): boolean {
```
Comment (mario-paniccia) [inline src/app/shared/case-editor/case-edit-submit.component.ts]: not sure I got what this method is doing. Why we need to call this.summaryCaseField(field)?

File: src/app/shared/case-editor/case-edit-submit.html [hunk 2]
```diff
@@ -28,13 +33,18 @@ <h2 class="plain-message">Check the information below carefully. This will form
             <ng-container *ngIf="canShowFieldInCYA(field)">
               <tr ccdConditionalShow ccdLabelSubstitutor [caseField]="field"
                 [formGroup]="editForm.controls['data']" [eventFields]="eventTrigger.case_fields">
-                <th class="valign-top">{{field.label}}</th>
+                <th class="valign-top"><span class="text-16">{{field.label}}</span></th>
                 <td class="form-cell">
-                  <ccd-field-read [caseField]="summaryCaseField(field)" [context]="paletteContext"></ccd-field-read>
+                  <a href="mailto:{{summaryCaseField(field).value}}" class="text-16" *ngIf="isEmail(field)">
```
Comment (mario-paniccia) [inline src/app/shared/case-editor/case-edit-submit.html]: this should be changed. It's not responsibility of case-edit-submit.html to check whether a field has an email and display that accordingly. 
This should be moved inside the email read component template

File: src/app/shared/case-editor/case-edit-submit.component.ts (hunk 1 above)
Comment (MKaramath) [reply src/app/shared/case-editor/case-edit-submit.component.ts]: I have fixed this by assigning it to a variable now.   I am calling this.summaryCaseField(field) to filter non-null fields.

File: src/app/shared/case-editor/case-edit-submit.html (hunk 2 above)
Comment (MKaramath) [reply src/app/shared/case-editor/case-edit-submit.html]: I tried but it was bit challenging due to that all types of read only fields are rendering dynamically.  May be I will look at that to improve sometimes in future if I find a better solution.

Comment (mario-paniccia) [reply src/app/shared/case-editor/case-edit-submit.html]: becasuse theis field is used only on the check your answers section of this page, we should rename it to ccd-cya-field-read 
and we should move all this logic inside it's template. We should also move the isEmail method inside the ccdCYAFieldRead component
Karamath agree that he's going to open a new ticket to do this change

**Comment by ngul** [comment]
This PR is absolutely massive. could there anything be done to split it?

**Pull request merged**